src/GovukNotify/Client/BaseClient.cs

```diff
@@ -34,55 +35,71 @@ public BaseClient(IHttpClient client, string apiKey, string baseUrl = "https://a
-            return await MakeRequest(url, HttpMethod.Post, content).ConfigureAwait(false);
+            using var content = new StringContent(json, Encoding.UTF8, "application/json");
+            return await MakeRequest(url, HttpMethod.Post, content, cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
         }
 
-        public async Task<byte[]> GETBytes(string url)
+        public async Task<byte[]> GETBytes(string url, CancellationToken cancellationToken = default)
         {
-            return await MakeRequestBytes(url, HttpMethod.Get).ConfigureAwait(false);
+            return await MakeRequestBytes(url, HttpMethod.Get, cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
         }
 
-        public async Task<byte[]> MakeRequestBytes(string url, HttpMethod method, HttpContent content = null) {
-            var response = SendRequest(url, method, content).Result;
-
-            var responseContent = await response.Content.ReadAsByteArrayAsync().ConfigureAwait(false);
-
+        public async Task<byte[]> MakeRequestBytes(string url, HttpMethod method, HttpContent content = null, CancellationToken cancellationToken = default) {
+            var response = await SendRequest(url, method, content, cancellationToken);
+
+#if NET6_0_OR_GREATER
+            var responseContent = await response.Content.ReadAsByteArrayAsync(cancellationToken)
+                .ConfigureAwait(false);
+#else
+            var responseContent = await response.Content.ReadAsByteArrayAsync()
+                .ConfigureAwait(false);
+#endif
+
             if (!response.IsSuccessStatusCode)
             {
                 // if there was an error, rather than a binary pdf, the http body will be a json error message, so
                 // encode the bytes as UTF8
-                HandleHTTPErrors(response, Encoding.UTF8.GetString(responseContent));
+                BaseClient.HandleHTTPErrors(response, Encoding.UTF8.GetString(responseContent));
             }
-            return responseContent;
 
+            return responseContent;
         }
 
-        public async Task<string> MakeRequest(string url, HttpMethod method, HttpContent content = null)
+        public async Task<string> MakeRequest(string url, HttpMethod method, HttpContent content = null, CancellationToken cancellationToken = default)
         {
-            var response = SendRequest(url, method, content).Result;
-
-            var responseContent = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
-
+            var response = await SendRequest(url, method, content, cancellationToken);
+
+#if NET6_0_OR_GREATER
+            var responseContent = await response.Content.ReadAsStringAsync(cancellationToken)
+                .ConfigureAwait(false);
+#else
+            var responseContent = await response.Content.ReadAsStringAsync()
+                .ConfigureAwait(false);
+#endif
```

Comment:
Again, as before this conditional compilation is required due to the target framework.